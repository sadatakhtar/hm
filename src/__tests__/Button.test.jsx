import React from "react";
import { render, fireEvent } from "@testing-library/react";
import Button from "../components/Button";

describe("Button component", () => {
  test("renders CustomButton correctly", () => {
    const { asFragment } = render(<Button text="Click Me" />);
    expect(asFragment()).toMatchSnapshot();
  });

  test("displays the correct text", () => {
    const { getByText } = render(<Button text="Submit" />);
    expect(getByText("Submit")).toBeInTheDocument();
  });
});
