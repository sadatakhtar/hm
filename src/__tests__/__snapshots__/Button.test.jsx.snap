// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component renders CustomButton correctly 1`] = `
<DocumentFragment>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary css-1abvrni-MuiButtonBase-root-MuiButton-root"
    tabindex="0"
    type="button"
  >
    Click Me
  </button>
</DocumentFragment>
`;
